test/integration/web_console_access_test.go

```diff
@@ -99,6 +119,73 @@ func TestAccessDisabledWebConsole(t *testing.T) {
 		"console/java":        {http.StatusForbidden, ""},
 	} {
 		url := masterOptions.AssetConfig.MasterPublicURL + "/" + endpoint
-		tryAccessURL(t, url, exp.statusCode, exp.location)
+		tryAccessURL(t, url, exp.statusCode, exp.location, nil)
+	}
+}
+
+func TestAccessOriginWebConsoleMultipleIdentityProviders(t *testing.T) {
+	masterOptions, err := testserver.DefaultMasterOptions()
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	// Replace the default IdentityProvider with an AllowAll provider
+	masterOptions.OAuthConfig.IdentityProviders[0] = configapi.IdentityProvider{
+		Name:            "foo",
+		UseAsChallenger: true,
+		UseAsLogin:      true,
+		MappingMethod:   "claim",
+		Provider:        &configapi.AllowAllPasswordIdentityProvider{},
+	}
+
+	// Set up a second AllowAll provider
+	masterOptions.OAuthConfig.IdentityProviders = append(masterOptions.OAuthConfig.IdentityProviders, configapi.IdentityProvider{
+		Name:            "bar",
+		UseAsChallenger: true,
+		UseAsLogin:      true,
+		MappingMethod:   "claim",
+		Provider:        &configapi.AllowAllPasswordIdentityProvider{},
+	})
```

Comment:
add a third one with a name like "baz qux" to make sure we URL-encode the idp param and login path segment correctly